service/src/main/java/org/broadinstitute/listener/relay/inspectors/SetDateAccessedInspector.java

```diff
@@ -88,10 +91,32 @@ public boolean inspectWebSocketUpgradeRequest(
     return checkLastAccessDateAndCallServiceIfExpired(relayedHttpListenerRequest);
   }
 
+  /**
+   * Inspect the request. Call Leonardo to updateDateAccessed on our resource, unless the request
+   * headers include {@code X-SetDateAccessedInspector-Action=ignore}.
+   *
+   * @return whether to relay the original request.
+   */
   @Override
   public boolean inspectRelayedHttpRequest(RelayedHttpListenerRequest relayedHttpListenerRequest) {
+    if (isActionIgnore(relayedHttpListenerRequest)) {
+      logger.info("SetDateAccessedInspector will ignore this request.");
+      return true;
```

Comment:
Might be a java newbie question, but why return true here?

Reply:
`inspectRelayedHttpRequest()` returns a bool which is used by the inspectors pipeline: all inspectors must return `true` in order for the request to be forwarded to the target. `SetDateAccessedInspector` never impedes requests; its action is either a no-op (ignore, this case) or a message to Leo to updateDateAccessed (the other case). In both cases, returning `true` just OKs the request to proceed.

Reply:
I could have moved the boolean return outside of the if-else block, since it should always return true, but having it explicit for each case seemed slightly more readable to a future dev (who might otherwise see the ignore log and think that an early-return might have been overlooked).